bindata/v3.11.0/openshift-apiserver/deploy.yaml

```diff
@@ -134,12 +134,12 @@ spec:
           periodSeconds: 5
           timeoutSeconds: 10
           successThreshold: 1
-          failureThreshold: 1
+          failureThreshold: 4
```

Comment:
@liouk, @benluddy left me this message:

```
so, the worst-case assumption is that the apiserver container gets sigterm from kubelet the instant after a successful readiness probe response
so the first post-shutdown-delay readiness probe fires after the probe's interval has elapsed
then kubelet needs to observe <failureThreshold> probe failures before becoming unready
each probe should take between 0 and <timeout> seconds to get a result
i think this means we need to wait <failureThreshold> * (<interval> + <timeout>) to become unready, starting from the time graceful shutdown begins
as the minimum
then we need some buffer on top of that for kubelet to update the pod ready condition, endpoint controller to observe the pod change and reconcile endpoints, and kube-apiserver endpoint informers to observe that change and update the lister caches
```
I think Ben is right. I would lover the `failureThreshold` to `3` and that would give us 3*(5+10) + 15 (time for going unready, removing the endpoint and for kas to observer the change) for shutdown-delay.

Reply:
As explained above, setting it to 3 would give us `5s + 3*10s` so if we add `15s` as a buffer, we get `50s` in total.